### PR TITLE
fix: stop divide by zero warning in LU solvers

### PR DESCRIPTION
**Pull request by wd15**
Prompted by [this](https://stackoverflow.com/questions/66163197/solving-a-pde-with-1d-zero-gradient-boundary-condition-and-zero-flux-condition-w/66163944#66163944).

### CODE REVIEW

**Comment by wd15** [comment]
Test failures seem unrelated to changes

File: shell.nix [hunk 1]
```diff
@@ -31,7 +31,12 @@ in
 
       export OMPI_MCA_plm_rsh_agent=/usr/bin/ssh
 
-      ## To build the docs
+      ## To build the docs (don't use --pure)
```
Comment (guyer) [inline shell.nix:34]: Should this go in #791?

Comment (wd15) [reply shell.nix]: Done

**Comment by wd15** [comment]
VTK failure on Travis is probably version incompatibility with Numpy.

**Comment by guyer** [comment]
I agree, the VTK errors are safe to ignore

**Comment by guyer** [comment]
I don't understand why the polyxtal example is stalling with SciPy on Py3k

**Comment by wd15** [comment]
> I don't understand why the polyxtal example is stalling with SciPy on Py3k

Is this the first time you've seen that?

**Comment by guyer** [comment]
I can't remember a stall like this for a serial run. I was wondering if the error was somehow negative, so that the `<=` test would fail, but I don't think that's possible and I don't know why that would make it stall.

**Comment by wd15** [comment]
> I can't remember a stall like this for a serial run. I was wondering if the error was somehow negative, so that the `<=` test would fail, but I don't think that's possible and I don't know why that would make it stall.

Not to worry. I'll try and debug it at some point. No rush to finish this.

**Comment by guyer** [comment]
polyxtal is stalling with SciPy on other branches. This doesn't have anything to do with it.

**Comment by wd15** [comment]
@guyer I merged from master and tests are failing for various reasons seemingly not related to the changes.

**Comment by guyer** [comment]
> @guyer I merged from master and tests are failing for various reasons seemingly not related to the changes.

Weird. Looks like a slew of unrelated failures. Should be a joy to debug.

**Comment by guyer** [comment]
Looks like a lot of things started failing [about 5 months ago](https://app.circleci.com/pipelines/github/usnistgov/fipy/326/workflows/b7a16b19-4b8e-4bc8-8734-d51d5aea4ada) when I relaxed the installation constraint on Gmsh from `"gmsh<4.0"` to just `gmsh`. (see #820)
